src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3093,29 +3093,33 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
 		// unconnected tcp sock is always ready for read!
 		// return its fd as ready
 		si_tcp_logdbg("block check on unconnected socket");
-		goto noblock_nolock;
+		return true;
 	}
 
 	if (m_n_rx_pkt_ready_list_count)
-		goto noblock_nolock;
+		return true;
 
 	if (!p_poll_sn)
 		return false;
 
 	consider_rings_migration();
 
-	m_rx_ring_map_lock.lock();
```

Comment:
Do we guarantee that `rx_add_ring_cb()` is not called during this function execution?
`rx_add_ring_cb()` can add new element into `m_rx_ring_map`

Reply:
?

Reply:
rx_add_ring_cb() can be called from another thread - this it the main idea behind this deadlock.
But the map is still protected - see line 3118